### DISCUSSION TITLE
[SECURITY] Option to block or restrict links

### DISCUSSION THREAD

**Discussion by bernd892374892** [ruffle-rs/ruffle#4303] [Ideas] [answered]
More than often flash apps contain external links - usually to credit the author. However, I've noticed that many domains have been taken over by unrelated parties and therefore clicking on them might be dangerous. Of course there are also maliciously crafted SWF files that do this on purpose.

What I propose is the following option:

`linkBehavior` A string, indicating how links should be processed. Possible values are:
- "normal": Ruffle always handles links normally by opening a new tab in the browser.
- "confirm": Ruffle always asks for confirmation before opening any link.
- "block": Ruffle always blocks every link.

**Comment by danielhjacobs** [comment]
This sounds like it ties in with the discussion that was happening on #3511. Your suggestion does provide an alternative approach, likely in addition to configurable blocked domains. I'm not really involved in Ruffle's development, but thought I'd point you to that PR so you could see the discussion that's taken place so far.

**Comment by n0samu** [comment, marked as answer]
This has been added in #10625 as `openUrlMode`. [See our wiki for more information](https://github.com/ruffle-rs/ruffle/wiki/Using-Ruffle#configuration-options).
